sdk/python/feast/infra/online_stores/online_store.py

```diff
@@ -0,0 +1,102 @@
+    """
+    OnlineStore is an object used for all interaction between Feast and the service used for online storage of
+    features.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def online_write_batch(
+        config: RepoConfig,
+        table: Union[FeatureTable, FeatureView],
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        """
+        Write a batch of feature rows to the online store. This is a low level interface, not
+        expected to be used by the users directly.
+
+        If a tz-naive timestamp is passed to this method, it should be assumed to be UTC by implementors.
+
+        Args:
+            config: The RepoConfig for the current FeatureStore.
+            table: Feast FeatureTable or FeatureView
+            data: a list of quadruplets containing Feature data. Each quadruplet contains an Entity Key,
+            a dict containing feature values, an event timestamp for the row, and
+            the created timestamp for the row if it exists.
+            progress: Optional function to be called once every mini-batch of rows is written to
+            the online store. Can be used to display progress.
+        """
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def online_read(
+        config: RepoConfig,
+        table: Union[FeatureTable, FeatureView],
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        """
+        Read feature values given an Entity Key. This is a low level interface, not
+        expected to be used by the users directly.
+
+        Args:
+            config: The RepoConfig for the current FeatureStore.
+            table: Feast FeatureTable or FeatureView
+            entity_keys: a list of entity keys that should be read from the FeatureStore.
+            requested_features: (Optional) A subset of the features that should be read from the FeatureStore.
+        Returns:
+            Data is returned as a list, one item per entity key. Each item in the list is a tuple
+            of event_ts for the row, and the feature data as a dict from feature names to values.
+            Values are returned as Value proto message.
+        """
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def setup(
+        config: RepoConfig,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
+        entities_to_delete: Sequence[Entity],
+        entities_to_keep: Sequence[Entity],
+        partial: bool,
+    ):
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def teardown(
+        config: RepoConfig,
+        tables: Sequence[Union[FeatureTable, FeatureView]],
+        entities: Sequence[Entity],
+    ):
```

Comment:
@woop I added these methods here (and ripped out stuff from `gcp.py` and `local.py`). please take a look!